src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -614,10 +616,100 @@ private static void AddObjectsIntoTargetCollection(object targetCollection, List
             }
         }
 
-        private static void SetMemberValue(object o, object value, string memberName)
+        private static IDictionary<Tuple<Type, Type>, Func<MemberInfo, SetMemberValueDelegate>> s_GetSetMemberValueDelegateWithTypeCache = new ConcurrentDictionary<Tuple<Type, Type>, Func<MemberInfo, SetMemberValueDelegate>>();
```

Comment:
Nit: Instead of typing the field as the IDictionary interface, consider strongly typing it as ConcurrentDictionary, to avoid interface dispatch when calling the dictionary. Same for the one below.